shader/draw_leaf.wgsl

```diff
@@ -151,8 +151,11 @@ fn main(
                 let r1 = bitcast<f32>(scene[dd + 6u]);
                 let inv_det = 1.0 / (matrx.x * matrx.w - matrx.y * matrx.z);
                 let inv_mat = inv_det * vec4(matrx.w, -matrx.y, -matrx.z, matrx.x);
-                var inv_tr = inv_mat.xz * translate.x + inv_mat.yw * translate.y;
-                inv_tr += p0;
+                var inv_tr = inv_det * vec2(
```

Comment:
Just to confirm my understanding, the 2x2 matrix is stored in column-major order in the vector, correct?

A slightly shorter formulation that reuses `inv_mat` and built-in matrix multiplication could look like:
```
let inv_tr = mat2x2<f32>(inv_mat.xy, inv_mat.zw) * (-translate) - p0;
```
which has the added benefit of declaring `inv_tr` as immutable.

Feel free to push back if you disagree (or if there's a performance assumption that I'm missing here). Otherwise the PR overall lgtm.

Reply:
I think the lack of matrix types in the codebase is simply a tradition at this point. I like this formulation and we should probably do more of it moving forward. Updated!